bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/formsportal/LinkImpl.java

```diff
@@ -143,20 +142,64 @@ public String getAssetPath() {
     }
 
     private Map<String, String> getQueryParams() {
-        if (queryParamsMap == null) {
+        if (queryParamsMap == null || this.queryParamsMap.isEmpty()) {
             populateQueryParams();
         }
         return queryParamsMap;
     }
 
     private void populateQueryParams() {
+        this.queryParamsMap = new HashMap<String, String>();
         if (paramsResourceList != null) {
-            this.queryParamsMap = new HashMap<String, String>();
             for (Resource param : paramsResourceList) {
                 ValueMap properties = param.getValueMap();
                 this.queryParamsMap.put(properties.get(PN_PARAM_KEY, ""),
                     properties.get(PN_PARAM_VALUE, ""));
             }
         }
     }
+
+    protected abstract static class FormsLinkProcessor implements Comparable<FormsLinkProcessor> {
```

Comment:
Why do you need to implement comparable here ?

Reply:
This is for priority based TreeMap.
Based on this comparator, TreeMap can automatically re-order the list of processors so that they always get iterated in a fixed order.